### ISSUE TITLE
Can you explain how interleave works?

### ISSUE THREAD

**Issue by thomascong121** [Prasanna1991/LatentMixing#3]
Hi, thank you for the work, I am a bit confused when I read your code about interleave. can you explain this part a bit more/
`
def interleave_offsets(batch, nu):
   groups = [batch // (nu + 1)] * (nu + 1)   
   for x in range(batch - sum(groups)):       
       groups[-x - 1] += 1    
       offsets = [0]    
       for g in groups:        
             offsets.append(offsets[-1] + g)    
       assert offsets[-1] == batch    
       return offsets

def interleave(xy, batch):    
      nu = len(xy) - 1    
      offsets = interleave_offsets(batch, nu)    
      xy = [[v[offsets[p]:offsets[p + 1]] for p in range(nu + 1)] for v in xy]    
      for i in range(1, nu + 1):        
           xy[0][i], xy[i][i] = xy[i][i], xy[0][i]    
      return [torch.cat(v, dim=0) for v in xy]
`
